### DISCUSSION TITLE
Prisma migration one to many incompatible types: text[] and text.

### DISCUSSION THREAD

**Discussion by 28development** [prisma/prisma#14008] [Q&A]
I updated my schema to add a many-to-many relation to my snippet model. 

This is how the part of the schema looks like

```ts
model Snippet {
  id           String           @id @default(uuid())
  title        String
  description  String
  content      String
  user         User             @relation(fields: [userId], references: [id])
  userId       String
  technology   Technology       @relation(fields: [technologyId], references: [id])
  technologyId String
  createdAt    DateTime         @default(now())
  updatedAt    DateTime         @updatedAt
  visibility   Visibility?      @relation(fields: [visibilityId], references: [id])
  visibilityId String?
  tags         TagsOnSnippets[]
}

model Tag {
  id        String           @id @default(uuid())
  name      String
  excerpt   String
  body      String
  createdAt DateTime         @default(now())
  updatedAt DateTime         @updatedAt
  snippets  TagsOnSnippets[]
}

model TagsOnSnippets {
  snippet   Snippet  @relation(fields: [snippetId], references: [id])
  snippetId String
  tag       Tag      @relation(fields: [tagId], references: [id])
  tagId     String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@id([snippetId, tagId])
}
```

If I try to run `npx prisma migrate dev` afterwards I get following error output:

Error: P3006

> Migration `20220623223220_update_relation_for_snippet_and_tags` failed to apply cleanly to the shadow database.
> Error:
> db error: ERROR: foreign key constraint "Snippet_tagId_fkey" cannot be implemented
> DETAIL: Key columns "tagId" and "id" are of incompatible types: text[] and text.
>    0: sql_migration_connector::validate_migrations
>              at migration-engine/connectors/sql-migration-connector/src/lib.rs:272
>    1: migration_core::state::DevDiagnostic
>              at migration-engine/core/src/state.rs:250

Not sure what I may be missing here.

**Comment by austincrim** [comment]
Hey @28development 👋,

What's the current state of your schema for `Tag`? Did you recently change the type of the id field?

If not, this seems unexpected and it would be great if you could file a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with more details.

**Reply by 28development** [reply]
Hi @austincrim ,

had no time yet to check this again, I am going to create one once I have time again. 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
